cli/lib/kontena/cli/stacks/yaml/service_extender.rb

```diff
@@ -12,7 +12,7 @@ def initialize(service_config)
       # @param [Hash] from
       # @return [Hash]
       def extend_from(from)
-        service_config['environment'] = extend_env_vars(from['environment'], service_config['environment'])
+        service_config['environment'] = extend_env_vars(from['env'], service_config['environment'])
```

Comment:
Surprised that the `extend` mechanic uses the generated service form... does that mean that things like `cmd` vs `command`, `memory` vs `mem_limit` etc are currently broken for extended services?

Separate issue... it would already be broken in the existing code, just noticed it here.